conn_test.go

```diff
@@ -837,12 +838,13 @@ func TestClientNewSessionInvalidSecondResponseSameChannel(t *testing.T) {
 	cancel()
 	require.NoError(t, err)
 	require.NotNil(t, session)
-	// second session fails
+	// second session fails - times out as the ack is never received
 	ctx, cancel = context.WithTimeout(context.Background(), 1*time.Second)
 	session, err = client.NewSession(ctx, nil)
 	cancel()
 	require.Error(t, err)
 	require.Nil(t, session)
+	require.ErrorIs(t, err, context.DeadlineExceeded)
 	require.NoError(t, client.Close())
```

Comment:
It appears there's a race someplace as this test failed in one of the runs.

```
2023-02-22T21:27:33.0779743Z     conn_test.go:848: 
2023-02-22T21:27:33.0780065Z         	Error Trace:	conn_test.go:848
2023-02-22T21:27:33.0780411Z         	Error:      	Received unexpected error:
2023-02-22T21:27:33.0781008Z         	            	*frames.PerformEnd: didn't find channel 0 in sessionsByRemoteChannel (PerformEnd)
2023-02-22T21:27:33.0781498Z         	Test:       	TestClientNewSessionInvalidSecondResponseSameChannel
```

Reply:
This is due to the shenanigans in `Session.begin()`.  In some cases the goroutine to end the session never runs, so we get a `context.DeadlineExceeded` error as expected.  In the other case, the goroutine does run; however, we're now sending an end performative to a peer that never received the corresponding begin performative.  I suppose in this case we should return an AMQP error.  That still doesn't fix the underlying race though.

Reply:
This has been resolved after rebasing on `main`.